src/server.c

```diff
@@ -351,6 +357,21 @@ uint64_t dictCStrCaseHash(const void *key) {
     return dictGenCaseHashFunction((unsigned char*)key, strlen((char*)key));
 }
 
+/* Dict hash function for client */
+uint64_t dictClientHash(const void *key) {
+    return ((client *)key)->id;
+}
+
+/* Dict compare function for client */
+int dictClientKeyCompare(dict *d, const void *key1, const void *key2) {
+    uint64_t l1,l2;
+    UNUSED(d);
+
+    l1 = ((client *)key1)->id;
+    l2 = ((client *)key2)->id;
+    return l1 == l2;
```

Comment:
Don't see much value in declaring those variables.
```suggestion
    UNUSED(d);
    return ((client *)key1)->id == ((client *)key2)->id;
```